src/cr/cube/matrix/assembler.py

```diff
@@ -1161,7 +1175,29 @@ def _subtotal_values(self):
         return intersections[:, self._insertion_idx]
 
 
-class _SortRowsByMarginalHelper(_RowOrderHelper):
+class _SortRowsByLabelHelper(_SortRowsBaseHelper):
```

Comment:
I think this looks much cleaner now. Basically every different type needs to compute element-values and subtotal-values, and everything else is handled but its superclasses. This feels even cleaner than I thought. Makes me want to invent a new sort-by-value type just to take advantage of that :)

Reply:
The customers are clearly asking for a `_SortRowsByRandomValueHelper` 😉